src/api/SFA.DAS.Provider.Events.Infrastructure/Data/DcfsTransferRepository.cs

```diff
@@ -11,68 +13,92 @@ namespace SFA.DAS.Provider.Events.Infrastructure.Data
-                includeIf: academicYear.HasValue && collectionPeriod.HasValue);
-
-            sqlBuilder.Where(
-                "TransferSenderAccountId = @senderAccountId",
-                new {senderAccountId},
-                includeIf: senderAccountId.HasValue);
-
-            sqlBuilder.Where(
-                "AccountId = @receiverAccountId",
-                new {receiverAccountId},
-                includeIf: receiverAccountId.HasValue);
-        }
+        private const string CountSqlTemplate = @"SELECT COUNT(1) [TotalCount] FROM [Payments2].[Payment] P /**where**/";
 
         public async Task<PageOfResults<TransferEntity>> GetTransfers(
             int page, int pageSize,
             long? senderAccountId = null,
             long? receiverAccountId = null,
             int? academicYear = null, 
             int? collectionPeriod = null)
+        {
+            using (var connection = await GetOpenConnection().ConfigureAwait(false))
+            {
+                var result = await GetUnPaginatedTransfers(page, pageSize, senderAccountId, receiverAccountId, academicYear, collectionPeriod, connection);
+
+                var count = await GetTransferCount(senderAccountId, receiverAccountId, academicYear, collectionPeriod, connection);
+
+                var pagedResults = PageResults(result.ToList(), page, pageSize, count);
+
+                return pagedResults;
+            }
+        }
+
+        private async Task<IEnumerable<TransferEntity>> GetUnPaginatedTransfers(int page, int pageSize, long? senderAccountId, long? receiverAccountId, int? academicYear, int? collectionPeriod, SqlConnection connection)
         {
             var sqlBuilder = new SqlBuilder();
             var query = sqlBuilder.AddTemplate(SqlTemplate);
 
             BuildQueryParameters(senderAccountId, receiverAccountId, academicYear, collectionPeriod, sqlBuilder);
-            sqlBuilder.AddParameters(new {PageIndex = page, PageSize = pageSize});
 
-            var result = await Query<TransferEntity>(query.RawSql, query.Parameters).ConfigureAwait(false);
-            var pagedResults = PageResults(result.ToList(), page, pageSize);
-            return pagedResults;
+            AddPagingQueryParameters(page, pageSize, sqlBuilder);
+
+            var result = await connection.QueryAsync<TransferEntity>(query.RawSql, query.Parameters).ConfigureAwait(false);
+
+            return result;
+        }
+
+        private static async Task<int> GetTransferCount(long? senderAccountId, long? receiverAccountId, int? academicYear, int? collectionPeriod, SqlConnection connection)
+        {
+            var countSQlBuilder = new SqlBuilder();
+            var countQuery = countSQlBuilder.AddTemplate(CountSqlTemplate);
+
+            BuildQueryParameters(senderAccountId, receiverAccountId, academicYear, collectionPeriod, countSQlBuilder);
+
+            var count = await connection.QueryFirstOrDefaultAsync<int>(
```

Comment:
You can use `ExecuteScalarAsync` to get the single returned value

Reply:
updated here and in payments repository